statsd/client.go

```diff
@@ -304,13 +303,24 @@ func NewClient(addr, prefix string) (Statter, error) {
 	return client, nil
 }
 
-// helper method to ensure a dot is added only when necessary
-func dotprefix(prefix string) string {
-	if prefix != "" && !strings.HasPrefix(prefix, ".") {
-		return "." + prefix
+// joinscopes is a helper that ensures we combine scopes with a dot when
+// it's appropriate to do so. Specifically this looks for trailing dots on
+// the existing prefix and leading dots on the new suffix.
+//
+// It returns the new prefix.
+func joinscopes(curPrefix, suffix string) string {
+	var (
+		emptyBase      = curPrefix == ""
+		baseHasDot     = strings.HasSuffix(curPrefix, ".")
```

Comment:
I don't think a test for current base prefix having a trailing dot is required. If that is present, it would mess up the actual metric generation itself. I think I originally tested for suffix having a leading dot, to avoid possible confusion with people not knowing if a dot was added for your or not. I don't think there is much confusion over whether the trailing dot on a base prefix should be present (it should not).

As a result, I think this can be simplified to:
~~~
suffix = strings.TrimLeft(suffix, ".")
if prefix != "" && suffix != "" {
    return prefix + "." + suffix
}
return prefix + suffix
~~~